File: 04 - ORIENTAÇÃO A OBJETOS/nodeJS - JavaScript_/Exercicio 2/Exercicio 2.js
class CaixaDeNumero {
    constructor(numeroSecreto){
        this.numeroSecreto = numeroSecreto
    }

    ehPar = () => this.numeroSecreto%2==0 ? 'É par' : 'É ímpar'

    dizerTabuada(){
        let vetor = []
        for(let i=0; i<10; i++){
            vetor[i] = this.numeroSecreto * (i+1)
        }
        return vetor
    }

    dizerNumeroSecreto = () => this.numeroSecreto

    estaEntre(inicio, fim){
        if(inicio>=this.numeroSecreto && fim<=this.numeroSecreto)
            console.log('Está dentro do intervalo informado')
        else
            console.log('Não está dentro do intervalo')
    }

    arriscaPalpite(palpite){
        if(palpite == this.numeroSecreto)
            console.log('PARABÉNS, VOCÊ ACERTOU!!')
        else
            console.log(`Infelizmente não é o ${palpite}, tente novamente`)
    }
}

function teste(incio, fim, palpite){
    var numeroSecreto = Math.floor(Math.random() * 10)
    let caixa = new CaixaDeNumero(numeroSecreto)
    var op=1
    do{
        switch(op){
            case 1:
                console.log(caixa.ehPar())
                break
            case 2:
                caixa.estaEntre(incio, fim)
                break
            case 3:
                caixa.arriscaPalpite(palpite)
                break
            case 4:
                console.log('A tabuada: '+caixa.dizerTabuada())
                break
            case 5:
                console.log('O número secreto é: '+caixa.dizerNumeroSecreto())
                break
            case 6:
                console.log('Saindo...\n')
                break
            default:
                console.log('Opção incorreta!')
                break
        }
        op++
    } while(op != 7)
}

teste(2, 6, 4)
teste(1, 5, 8)

File: 04 - ORIENTAÇÃO A OBJETOS/Python/Exercicio 2/Exercicio 2.py
class CaixaDeNumero:
    def __init__(self, numeroSecreto):
        self.numeroSecreto = numeroSecreto

    def ehPar(self):
        return self.numeroSecreto % 2 == 0

    def dizerTabuada(self):
        array = []
        for i in range(10):
            array.append(self.numeroSecreto * (i+1))
        return array

    def dizerNumero(self):
        print('O número secreto é: {}'.format(self.numeroSecreto))

    def estaEntre(self, inicio, fim):
        if (inicio <= self.numeroSecreto and fim >= self.numeroSecreto):
            print('Está entre o intervalo informado')
        else:
            print('Não está entre o intervalo informado')
    
    def dizerPalpite(self, palpite):
        if palpite == self.numeroSecreto:
            print('PARABÉNS VOCÊ ACERTOU')
        else:
            print('Infelizmente não é esse tente novamente')

import random
#import os
numero = random.randrange(1, 10)
caixa = CaixaDeNumero(numero)

while True:
    print('\n1 - Verificar se é par')
    print('2 - Verificar se o numero se encontra em um intervalo')
    print('3 - Arriscar um palpite')
    print('4 - Dizer a tabuada')
    print('5 - INFORMAR O NÚMERO')
    print('6 - sair')
    opcao = int(input('Opção >> '))

    if opcao == 1:
        if caixa.ehPar():
            print('O numero é par')
        else:
            print('O número é ímpar')

    elif opcao == 2:
        inicio = int(input('Digite um intervalo\ndgite o ponto de inicio: '))
        fim = int(input('Digite o ponto de fim: '))
        print(caixa.estaEntre(inicio, fim))

    elif opcao == 3:
        palpite = int(input('Digite um palpite: '))
        caixa.dizerPalpite(palpite)

    elif opcao == 4:
        print(caixa.dizerTabuada())

    elif opcao == 5:
        caixa.dizerNumero()

    elif opcao == 6:
        print('SAINDO...')
        break

    else:
        print('Opção Inválida!')
    
    #os.system('pause')
    #os.system('cls') or None